distributed_nhc/distributed_nhc.sb.sh

```diff
@@ -0,0 +1,326 @@
+        echo "Nodefile $1 does not exist"
+        exit 1
+    fi
+
+    if [ -f "$1" ]; then
+        mapfile -t NODELIST_ARR < $1
+    fi
+    ;;
+-w|--nodelist) 
+    shift
+    if [ "$EXECUTION_MODE" == "SLURM" ]; then
+        echo "Cannot specify -w/--nodelist when running with SLURM, please pass node list to sbatch instead"
+        exit 1
+    fi
+
+    echo "Adding nodes from nodelist $1"
+    NODELIST_ARR+=( $(expand_nodelist $1 ) )
+    ;;
+# Shared Onetouch NHC Args
+-v|--version) 
+    shift
+    ONETOUCH_NHC_ARGS+=("-v" "$1")
+    ;;
+-g|--git) 
+    shift
+    ONETOUCH_NHC_ARGS+=("-g" "$1")
+    ;;
+-c|--config) 
+    shift
+    CUSTOM_CONF="$(realpath -e ${1//\~/$HOME})"
+    ONETOUCH_NHC_ARGS+=("-c" "$CUSTOM_CONF")
+    ;;
+-f|--force)
+    ONETOUCH_NHC_ARGS+=("-f")
+    ;;
+-V|--verbose)
+    ONETOUCH_NHC_ARGS+=("-V")
+    VERBOSE="True"
+    ;;
+# Shared Kusto Export Args
+--kusto-export-url) 
+    shift
+    echo "Setting Kusto export url to $1"
+    KUSTO_EXPORT_ENABLED="True"
+    KUSTO_EXPORT_ARGS+=("--ingest_url" "$1")
+    ;;
+--kusto-database) 
+    shift
+    echo "Setting Kusto database to $1"
+    KUSTO_EXPORT_ARGS+=("--database" "$1")
+    ;;
+--kusto-identity) 
+    shift
+    echo "Setting Kusto identity to $1"
+    KUSTO_IDENTITY="$1"
+    # Handle case of no client id provided
+    if [ -z "$KUSTO_IDENTITY" ]; then
+        echo "No client id provided, using system assigned identity"
+        KUSTO_IDENTITY="True"
+    fi
+    ;;
+--kusto-health-table) 
+    shift
+    echo "Setting Kusto health table to $1"
+    KUSTO_EXPORT_ARGS+=("--health_table_name" "$1")
+    ;;
+--kusto-debug-table)
+    shift
+    echo "Setting Kusto debug table to $1"
+    KUSTO_EXPORT_ARGS+=("--debug_table_name" "$1")
+    ;;
+--)
+    shift
+    break;;
+esac
+shift
+done
+
+if [ ${#NODELIST_ARR[@]} -eq 0 ]; then
+    echo "No nodes provided, must provide at least one node either from a file with -F/--nodefile or as a slurm node list with -w/--nodelist"
+    echo
+    print_help
+    exit 1
+fi
+
+NODELIST_ARR=( $(echo "${NODELIST_ARR[@]}" | tr ' ' '\n' | sort -u | tr '\n' ' ') )
+
+if [ $EXECUTION_MODE == "SLURM" ]; then
+    # Running with SLURM
+    { RAW_OUTPUT=$(srun --gpus-per-node=8 $ONETOUCH_NHC_PATH -n $NHC_JOB_NAME $@ | tee /dev/fd/3 ); } 3>&1
+else
+    # Running with Parallel SSH
+
+    # Log file paths
+    output_path="logs/$NHC_JOB_NAME.out"
+    error_path="logs/$NHC_JOB_NAME.err"
+
+    # Pssh args
+    timeout=900 # 15 minute timeout
+    
+    pssh_host_args=()
+    for node in "${NODELIST_ARR[@]}"; do
+        pssh_host_args+="-H $node "
+    done
+
+    echo "Running Parallel SSH Distributed NHC on:" 
+    echo "${NODELIST_ARR[@]}" | tr ' ' '\n' 
+    echo "======================"
+    echo "The health check is running, it will take a few minutes to complete."
+    RAW_OUTPUT=$(parallel-ssh -P -t $timeout ${pssh_host_args[@]} $ONETOUCH_NHC_PATH ${ONETOUCH_NHC_ARGS[@]} 3> $error_path | tee $output_path)
```

Comment:
Should there be a way to run the tests skipping the setup steps built in to onetouch_nhc.sh?

Reply:
onetouch_nhc.sh does already have a built-in mechanism to skip the build steps if it detects all the pre-requisites are already installed. It should be skipping the setup steps after the first run on a fresh cluster or even skipping them if ./install-nhc.sh was ran on a node already outside of distributed NHC (which installs the same set of pre-reqs)